app/src/main/java/com/github/houseorganizer/houseorganizer/image/QRAnalyzer.java

```diff
@@ -0,0 +1,47 @@
+package com.github.houseorganizer.houseorganizer.image;
+
+import static android.graphics.ImageFormat.YUV_420_888;
+import static android.graphics.ImageFormat.YUV_422_888;
+import static android.graphics.ImageFormat.YUV_444_888;
+
+import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.ImageProxy;
+
+import com.google.zxing.BinaryBitmap;
+import com.google.zxing.PlanarYUVLuminanceSource;
+import com.google.zxing.Result;
+import com.google.zxing.common.HybridBinarizer;
+import com.google.zxing.multi.qrcode.QRCodeMultiReader;
+
+import java.nio.ByteBuffer;
+
+public class QRAnalyzer implements ImageAnalysis.Analyzer {
```

Comment:
Could you make this class final & the same for its listener attribute?

Reply:
done

Reply:
Nice, thanks!